src/main/peripherals/Lock.java

```diff
@@ -0,0 +1,55 @@
+package main.peripherals;
+import java.io.IOException;
+import java.net.InetAddress;
+
+public class Lock implements IPeripheral{
+	private int id;
+	
+	private String ip;
+	private String port;
+	
+	public Lock(String ip, String port, int id){
+		this.id = id;
+		this.ip = ip;
+		this.port = port;
+	}
+	
+	public Lock(InetAddress ip, int port, int id){
```

Comment:
A better implementation, with less duplicatecode would be
Lock(ip,toString(), port, id).
This way there is less duplicate code.